client/boot/index.js

```diff
@@ -323,7 +323,11 @@ function reduxStoreReady( reduxStore ) {
 			} );
 
 			if ( '/' === context.pathname && config.isEnabled( 'devdocs/redirect-loggedout-homepage' ) ) {
-				page.redirect( '/devdocs/start' );
+				if ( config.isEnabled( 'oauth' ) ) {
+					page.redirect( '/authorize' );
+				} else {
+					page.redirect( '/devdocs/start' );
```

Comment:
> regular oauth should work for users that don't want to develop, and devdocs/start should be for contributors

I agree. This seems a good rule of thumb.

> If we want to cover the more generic case, does this flow make sense?

Sorry I had to rewrite it backward to process it – that's how I work:

* no oauth + no devdocs = WordPress.com login
* yes oauth + no devdocs = `/authorize`
* no oauth + yes devdocs = `/devdocs/start`
* yes oauth + yes devdocs = `/start/developer` 

In this structure, it seems to me that `devdocs` check is used as a proxy for "developer intent". In that case, I feel there should be a single landing page that deals with whichever state it reaches, and then provide follow-ups from that page.

So I guess what I mean is: 

* yes devdocs + any oauth = `/devdocs/start` (any developer)
* no devdocs + yes oauth = `/authorize` (proper OAuth)
* no devdocs + no oauth = WordPress.com login (fallback)

It's almost identical to what you outlined I think, the only difference is deferring anything dev related to a separate page (which then can deal with that as needed).